src/rewrite.rs

```diff
@@ -38,6 +56,12 @@ pub(crate) enum RewriteError {
     #[error("It exceeds the required width of {configured_width} for the span: {span:?}")]
     ExceedsMaxWidth { configured_width: usize, span: Span },
 
+    #[error("Failed to format given macro{} at: {span:?}", macro_error_kind)]
+    MacroFailure {
+        macro_error_kind: MacroErrorKind,
+        span: Span,
+    },
+
```

Comment:
I modeled `MacroErrorKind` as a field of `MacroFailure` struct rather than directly enumerating these under RewriteError. Thereby we won't need to change code matching whole RewriteError when we only modify macro related errors.

Reply:
maybe we can name `macro_error_kind` as `kind` to keep this consistent with other nodes like `rustc_ast::ast::ExprKind`

Reply:
I renamed it, thank you for suggestion!